mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/server/master/persistence/KeyValueBasedPersistenceProvider.java

```diff
@@ -689,6 +691,15 @@ public List<JobArtifact> listJobArtifacts(String name, String version) throws IO
             }).filter(Objects::nonNull).collect(Collectors.toList());
     }
 
+    @Override
+    public List<URI> listJobArtifactsToCache(ClusterID clusterID) throws IOException {
```

Comment:
how does these get updated? I don't see a write path.

Reply:
Right now manually. will add a write path on another PR so we can move on with the migration

Reply:
I guess this only returns a list for caching, and write path is request handling?

Reply:
@liuml07 yeah I am referring to that there is no API to write/update this list.